tst/EnergyPlus/unit/OutputProcessor.unit.cc

```diff
@@ -3167,8 +3172,148 @@ namespace EnergyPlus {
+			}
+			TimeValue.allocate( 2 );  // should this be commented out since belong to anonymous namespace in different file?
+			auto timeStep = 1.0 / 6;
+			SetupTimePointers( "Zone", timeStep );
+			SetupTimePointers( "HVAC", timeStep );
+
+			SetTimeValueMinutes( 10, 10 );
+
+
+			DataGlobals::WarmupFlag = true;
+
+			SetOutputProcessorReportNumberCounter( 0 );
+			ReportOutputFileHeaders();
+
+			PurchAir( 1 ).TotHeatEnergy = 1.1;
+			UpdateMeterReporting();
+			UpdateDataandReport( DataGlobals::HVACTSReporting );
+
+			PurchAir( 1 ).TotHeatEnergy = 1.3;
+			UpdateMeterReporting();
+			UpdateDataandReport( DataGlobals::HVACTSReporting );
+
+			PurchAir( 1 ).TotHeatEnergy = 1.5;
+			UpdateMeterReporting();
+			UpdateDataandReport( DataGlobals::HVACTSReporting );
+
+			PurchAir( 1 ).TotHeatEnergy = 1.7;
+			UpdateMeterReporting();
+			UpdateDataandReport( DataGlobals::HVACTSReporting );
+
+			PurchAir( 1 ).TotHeatEnergy = 1.9;
+			UpdateMeterReporting();
+			UpdateDataandReport( DataGlobals::HVACTSReporting );
+
+			PurchAir( 1 ).TotHeatEnergy = 2.2;
+			UpdateMeterReporting();
+			UpdateDataandReport( DataGlobals::HVACTSReporting );
+
+			DataGlobals::WarmupFlag = false;
+
+			PurchAir( 1 ).TotHeatEnergy = 2.4;
+			UpdateMeterReporting();
+			UpdateDataandReport( DataGlobals::ZoneTSReporting ); //zone timestep              
+
+
+			compare_eso_stream( delimited_string( {
+				"1,1,,Zone Ideal Loads Supply Air Total Heating Energy [J] !Each Call",
+				"32,11,,Zone Ideal Loads Supply Air Total Heating Energy [J] !RunPeriod [Value,Min,Month,Day,Hour,Minute,Max,Month,Day,Hour,Minute]",
+				"2,365,12,31, 0,24,10.00,20.00,Tuesday",
+				"1,1.1",
+				"2,365,12,31, 0,24,20.00,30.00,Tuesday",
+				"1,1.3",
+				"2,365,12,31, 0,24,30.00,40.00,Tuesday",
+				"1,1.5",
+				"2,365,12,31, 0,24,40.00,50.00,Tuesday",
+				"1,1.7",
+				"2,365,12,31, 0,24,50.00,60.00,Tuesday",
+				"1,1.9",
+				"2,365,12,31, 0,24,60.00,70.00,Tuesday",
+				"1,2.2",
+				"5,365",
+				"32,9.7,1.1,12,31,24,20,2.2,12,31,24,70",
+			} ) );
+
+
+			// must call the function twice to perform reset
+			DataGlobals::WarmupFlag = true;
+			ResetAccumulationWhenWarmupComplete( DataGlobals::WarmupFlag );
+			DataGlobals::WarmupFlag = false;
+			ResetAccumulationWhenWarmupComplete( DataGlobals::WarmupFlag );
+
+			PurchAir( 1 ).TotHeatEnergy = 100.0;
+			UpdateMeterReporting();
+			UpdateDataandReport( DataGlobals::HVACTSReporting );
+
+			PurchAir( 1 ).TotHeatEnergy = 200.0;
+			UpdateMeterReporting();
+			UpdateDataandReport( DataGlobals::HVACTSReporting );
+			
+			PurchAir( 1 ).TotHeatEnergy = 300.0;
+			UpdateMeterReporting();
+			UpdateDataandReport( DataGlobals::ZoneTSReporting ); //zone timestep              
+
+			compare_eso_stream( delimited_string( {
+				"2,365,12,31, 0,24, 0.00,10.00,Tuesday",
+				"1,100.",
+				"2,365,12,31, 0,24,10.00,20.00,Tuesday",
+				"1,200.",
+				"5,365",
+				"32,300.,100.,12,31,24,10,200.,12,31,24,20",
+			} ) );
+
+
+			PurchAir.deallocate();
```

Comment:
@JasonGlazer Since PurchAir is a local variable, you do not need to call deallocate() explicitly. It will fall out of scope at the end of this unit test and deallocate() itself. We needed to call deallocate() in unit tests for Arrays in the global namespaces since these do NOT fall out of scope at the end of each unit test. That said, calling deallocate() here isn't hurting anything.